Calculation of diagonals of cov matrix

jenke0 opened the pull request.

jenke0 committed:
add diagonals of c matrix

jnhewitt commented on src/direct_optimal_mapping/horizon_map.py:
```diff
@@ -274,6 +274,25 @@ def calc_map(self, uvw_sign=1, buffer=True, return_b1map=False,
+            if self.return_pmatrix == False:
+                print("return_pmatrix must be True")
+                break
+            self.pmatrix = self.pmatrix.reshape(len(self.pmatrix[0][0][0])**2, len(self.pmatrix[0][0][0])**2)
```
The two dimensions of the P-matrix in general are not the same.

jenke0 committed:
fix d mat

- remove fake a mat
- make d matrix not be a huge matrix to decrease disk space
- change "noise_map" to "nsig"
- add cmap to 'if' statements so you don't need to have return_pmat to have enough calculated for this slow calculation
- add in jackie's lines to reshape the p matrix

jnhewitt committed:
Merge branch 'main' into cmat-slow
